runtime/lua/vim/pack.lua

```diff
@@ -0,0 +1,911 @@
+end
+
+--- @param plug_list vim.pack.PlugList
+--- @param opts { skip_same_sha: boolean }
+--- @return string[]
+local function compute_feedback_lines(plug_list, opts)
+  -- Construct plugin line groups for better report
+  local report_err, report_update, report_same = {}, {}, {}
+  for _, p in ipairs(plug_list.list) do
+    local group_arr = #p.job.err > 0 and report_err
+      or (p.info.sha_head ~= p.info.sha_target and report_update or report_same)
+    table.insert(group_arr, compute_feedback_lines_single(p))
+  end
+
+  local lines = {}
+  --- @param header string
+  --- @param arr string[]
+  local function append_report(header, arr)
+    if #arr == 0 then
+      return
+    end
+    header = header .. ' ' .. string.rep('─', 79 - header:len())
+    table.insert(lines, header)
+    vim.list_extend(lines, arr)
+  end
+  append_report('# Error', report_err)
+  append_report('# Update', report_update)
+  if not opts.skip_same_sha then
+    append_report('# Same', report_same)
+  end
+
+  return vim.split(table.concat(lines, '\n\n'), '\n')
+end
+
+--- @param plug_list vim.pack.PlugList
+local function feedback_log(plug_list)
+  local lines = compute_feedback_lines(plug_list, { skip_same_sha = true })
+  local title = string.format('========== Update %s ==========', get_timestamp())
+  table.insert(lines, 1, title)
+  table.insert(lines, '')
+
+  local log_path = vim.fn.stdpath('log') .. '/nvimpack.log'
+  vim.fn.mkdir(vim.fs.dirname(log_path), 'p')
+  vim.fn.writefile(lines, log_path, 'a')
+end
+
+local function show_confirm_buf(lines, opts)
+  -- Show buffer in a separate tabpage
+  local bufnr = api.nvim_create_buf(true, true)
+  api.nvim_buf_set_name(bufnr, 'nvimpack://' .. bufnr .. '/confirm-update')
+  api.nvim_buf_set_lines(bufnr, 0, -1, false, lines)
+  vim.cmd.sbuffer({ bufnr, mods = { tab = vim.fn.tabpagenr('#') } })
+  local tab_num, win_id = api.nvim_tabpage_get_number(0), api.nvim_get_current_win()
+
+  local delete_buffer = vim.schedule_wrap(function()
+    pcall(api.nvim_buf_delete, bufnr, { force = true })
+    pcall(vim.cmd.tabclose, tab_num)
+    vim.cmd.redraw()
+  end)
+
+  -- Define action on accepting confirm
+  local function finish()
+    opts.exec_on_write(bufnr)
+    delete_buffer()
+  end
+  -- - Use `nested` to allow other events (useful for statuslines)
+  api.nvim_create_autocmd('BufWriteCmd', { buffer = bufnr, nested = true, callback = finish })
+
+  -- Define action to cancel confirm
+  --- @type integer
+  local cancel_au_id
+  local function on_cancel(data)
+    if tonumber(data.match) ~= win_id then
+      return
+    end
+    pcall(api.nvim_del_autocmd, cancel_au_id)
+    delete_buffer()
+  end
+  cancel_au_id = api.nvim_create_autocmd('WinClosed', { nested = true, callback = on_cancel })
+
+  -- Set buffer-local options last (so that user autocmmands could override)
+  vim.bo[bufnr].modified, vim.bo[bufnr].modifiable = false, false
+  vim.bo[bufnr].buftype, vim.bo[bufnr].filetype = 'acwrite', 'nvimpack'
```

Comment:
Module name is `vim.pack`, but using 'vimpack' as file/filetype name feels wrong (as it doesn't use 'nvim'). So it is 'nvimpack' for now.